hircluster.c

```diff
@@ -1263,12 +1263,11 @@ static int cluster_update_route_by_addr(redisClusterContext *cc, const char *ip,
         goto error;
     }
 
-    if (cc->connect_timeout) {
-        c = redisConnectWithTimeout(ip, port, *cc->connect_timeout);
-    } else {
-        c = redisConnect(ip, port);
-    }
+    redisOptions options = {0};
```

Comment:
Maybe this should have been put in a separate PR

Reply:
Doesn't matter, but we should fix the build before merging. Add an explicit cast or so, I guess.

Reply:
I thought this was possible from c99, maybe its some kind of gcc extension not available in early clang.
Anyway its fixable.

Reply:
Probably the old clang didn't follow c99 correctly.

Redis itself compiles with the following warning flags: `-Wall -W -Wno-missing-field-initializers`.

Reply:
Ah interesting, looked at hiredis which also have it:
`-Wall -W -Wstrict-prototypes -Wwrite-strings -Wno-missing-field-initializers`

but its not defined in its CMakeLists.txt weirdly enough.